### PR TITLE
[flutter_tools] support memory profiles from flutter drive

### LINKED ISSUE #82726: Devtools Memory tests should use the flutter tools distributed devtools
This would simplify the test integrations and help us avoid doing a pub global activate once we are using the dds distributed devtools.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/82726

I believe the devtools issues may be caused by a race between the run activated devtools and the test script activated devtools. Add support to flutter drive for capturing the memory profile directly.

~~Still WIP~~

### CODE REVIEW

**Comment by jonahwilliams** [comment]
So when i run this change, I get a completely empty memory profile. When I check out upstream/master I also get a completely empty memory profile. Did this test break?

**Comment by jonahwilliams** [comment]
Ahh found it:

```
[complex_layout_scroll_perf__devtools_memory] [STDOUT] stderr: [  +14 ms] Unhandled exception:
[complex_layout_scroll_perf__devtools_memory] [STDOUT] stderr: [        ] Exception: There are no registered methods for service "_flutter.listViews"
[complex_layout_scroll_perf__devtools_memory] [STDOUT] stderr: [        ] #0      MemoryProfile.callService (package:devtools_server/src/memory_profile.dart:90:7)
[complex_layout_scroll_perf__devtools_memory] [STDOUT] stderr: [        ] #1      MemoryProfile.getFlutterViewId (package:devtools_server/src/memory_profile.dart:219:43)
[complex_layout_scroll_perf__devtools_memory] [STDOUT] stderr: [        ] #2      MemoryProfile.getRasterCacheMetrics (package:devtools_server/src/memory_profile.dart:249:26)
[complex_layout_scroll_perf__devtools_memory] [STDOUT] stderr: [        ] #3      MemoryProfile._fetchRasterCacheInfo (package:devtools_server/src/memory_profile.dart:210:28)
[complex_layout_scroll_perf__devtools_memory] [STDOUT] stderr: [        ] #4      MemoryProfile._pollMemory (package:devtools_server/src/memory_profile.dart:190:25)
[complex_layout_scroll_perf__devtools_memory] [STDOUT] stderr: [        ] <asynchronous suspension>
[complex_layout_scroll_perf__devtools_memory] [STDOUT] stdout: [ +396 ms] 00:00 +0: scrolling performance test (setUpAll)
```

**Comment by jonahwilliams** [comment]
this test will still be a no-op until https://github.com/flutter/devtools/issues/3037 is fixed

File: packages/flutter_tools/lib/src/drive/drive_service.dart [hunk 1]
```diff
@@ -242,14 +256,30 @@ class FlutterDriverService extends DriverService {
...
-    ], environment: <String, String>{
-      'VM_SERVICE_URL': _vmServiceUri,
-      ...environment,
-    });
+    if (profileMemory != null) {
+      unawaited(_devtoolsLauncher.launch(
+        Uri.parse(_vmServiceUri),
+        additionalArguments: <String>['--profile-memory=$profileMemory'],
+      ));
+      // When profiling memory the original launch future will never complete.
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/drive/drive_service.dart:266]: Why is this? Does the process get leaked?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/drive/drive_service.dart:266]: The process doesn't leak (see below where we make sure to kill it), but it doesn't behave the same with the "devtools serving on X", instead it should start writing out a memory profile

**Review by zanderso — APPROVED** [review]

**Pull request merged**